src/app/services/access-guard.service.ts

```diff
@@ -0,0 +1,68 @@
+import { Injectable, Provider } from '@angular/core';
+import { CanActivate, ActivatedRouteSnapshot, RouterStateSnapshot, Router } from '@angular/router';
+import { ApiService } from './api.service';
+import { AuthService } from './auth.service';
+import { Observable } from 'rxjs/Observable';
+import 'rxjs/add/observable/of';
+import 'rxjs/add/observable/merge';
+import 'rxjs/add/operator/combineLatest';
+
+@Injectable()
+export class AccessGuard implements CanActivate {
+  constructor(
+    private apiService: ApiService,
+    private authService: AuthService,
+    private router: Router
+  ) { }
+
+  canActivate(route: ActivatedRouteSnapshot, state: RouterStateSnapshot): Promise<boolean> {
+    return new Promise((resolve) => {
+      let userId = null;
+      let userData: any = this.authService.getData();
+      if (userData) {
+        userId = userData.id;
+      }
+
+      if (route.url && route.url[0]) {
+        if (route.url[0].path === 'repo') {
```

Comment:
I would personally use `switch` here, so many nested else ifs are not so nice, imho. 
also, there may be more routes to come... what do you think?